rotkehlchen/chain/arbitrum_one/modules/umami/decoder.py

```diff
@@ -0,0 +1,175 @@
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Deposit {amount} {event.asset.symbol_or_name()} into Umami'
+            else:
+                event.event_type = HistoryEventType.SPEND
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.notes = f'Return {amount} {event.asset.symbol_or_name()} to Umami'
+
+        elif event.balance.amount == asset_normalized_value(
+                amount=context.transaction.value,
+                asset=event.asset.resolve_to_crypto_asset(),
+        ):  # Execution cost, balance will match transaction value
+            event.event_type = HistoryEventType.DEPOSIT if is_deposit else HistoryEventType.WITHDRAWAL  # noqa: E501
+            event.event_subtype = HistoryEventSubType.FEE
+            event.notes = f'Execution cost of {event.balance.amount} ETH'
+
+    def _decode_umami_vault_events(self, context: DecoderContext) -> DecodingOutput:
+        """Decode umami vault deposit/withdraw events.
+
+        A deposit/withdraw consists of two transactions, a request tx and an execution tx.
+        The event for deposit execution is not available in decoded_events here,
+        and must be transformed later via an ActionItem.
+        """
+        found_relevant_events = False
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                context.tx_log.topics[0] in {DEPOSIT_TOPIC, WITHDRAW_TOPIC}
+            ):  # deposit or withdraw request
+                self._decode_deposit_withdraw_request(context, event)
+                found_relevant_events = True
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                context.tx_log.topics[0] == TRANSFER_TOPIC
+            ):  # withdraw execution
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Receive {event.balance.amount} {event.asset.symbol_or_name()} from Umami'  # noqa: E501
+                event.counterparty = CPT_UMAMI
+                found_relevant_events = True
+
+        if found_relevant_events:
+            return DEFAULT_DECODING_OUTPUT
+
+        if (
+            context.tx_log.topics[0] == TRANSFER_TOPIC and
+            context.transaction.input_data.startswith(DEPOSIT_EXECUTION_FOUR_BYTES)
+        ):  # deposit execution
+            vault_asset = Asset(evm_address_to_identifier(
+                address=context.tx_log.address,
+                chain_id=self.evm_inquirer.chain_id,
+                token_type=EvmTokenKind.ERC20,
+            ))
+            receive_amount = asset_normalized_value(
+                amount=int.from_bytes(context.tx_log.data[0:32]),
+                asset=vault_asset.resolve_to_crypto_asset(),
```

Comment:
also resolve can happen above and then use resolved asset also in the action item